backend/agentd/watcher.go

```diff
@@ -7,67 +7,59 @@ import (
 	"github.com/coreos/etcd/clientv3"
 	"github.com/gogo/protobuf/proto"
 	corev3 "github.com/sensu/sensu-go/api/core/v3"
-	"github.com/sensu/sensu-go/backend/messaging"
 	"github.com/sensu/sensu-go/backend/store"
 	etcdstore "github.com/sensu/sensu-go/backend/store/etcd"
 	storev2 "github.com/sensu/sensu-go/backend/store/v2"
 	etcdstorev2 "github.com/sensu/sensu-go/backend/store/v2/etcdstore"
 	"github.com/sensu/sensu-go/backend/store/v2/wrap"
 )
 
-// EntityConfigWatcher watches changes to EntityConfig in etcd and publish them
+// GetEntityConfigWatcher watches changes to EntityConfig in etcd and publish them
 // over the bus as store.WatchEventEntityConfig
-func EntityConfigWatcher(ctx context.Context, client *clientv3.Client, bus messaging.MessageBus) {
+func GetEntityConfigWatcher(ctx context.Context, client *clientv3.Client) <-chan store.WatchEventEntityConfig {
 	key := etcdstorev2.StoreKey(storev2.ResourceRequest{
 		Context:   ctx,
 		StoreName: new(corev3.EntityConfig).StoreName(),
 	})
 	w := etcdstore.Watch(ctx, client, key, true)
+	ch := make(chan store.WatchEventEntityConfig, 1)
```

Comment:
My last question: why a buffer size of 1 and not, something bigger (but still reasonable, after defining that).
8? 32?

That would mean decoupling watch event reception and updating the bus even more, in case there is a spike in watch events or a temporary slow down on the bus side?

Reply:
Argh sorry I merged this too soon! I would be down to test this theory.